### ISSUE TITLE
ToDo mark all as complete is failing on in-memory implementation

### ISSUE THREAD

**Issue by thelordoftheboards** [codefoundries/isomorphic-material-relay-starter-kit#111]

**Comment by alexhawkins** [comment]
Why do you thinks it's failing? Have you also noticed that the app feels a bit sluggish? Should be light speed. I honestly think this might be an inherent issue with Material UI. Material UI feels slow to me. I recently built a fairly large app with redux and material ui, thought redux was this issue for the longest time. But now i'm starting think Material UI has some serious issues. Have you noticed how long it takes for checkboxes to animate after clicking on them multiple times?

**Comment by thelordoftheboards** [comment]
I am not quite sure yet, I know it did work before, I must have broken something while adding the user_id to all calls, or something like that. I will definitely look into that soon. I do not think this part is in any way related to material-ui.
I also added flow recently and there are some errors reported by flow, I have not gone through them yet, maybe there lays the answer.
Note to self - add tests :)

As for the speed I have not tested much on hardware other than my 2011 MBP, and on my machine it seemed speedy with normal usage. I also tested on iPhone 6. I have the feeling that the display happens later, after the click is performed, pending the animation. I tested on my iPhone to have a to do and:

* Double tap multiple times - the check status does not change, and I don't even see it flicking to the other state temporarily.
* Triple tap - the status always changes, but there does not seem to be on-off-on thing, it just changes to the next state.

I will play with this more. Could be also something to do with Relay. I will have to get my dev tools open and click furiously to see what happens and when.
